### DISCUSSION TITLE
Misorientation in mtex

### DISCUSSION THREAD

**Discussion by trudator** [mtex-toolbox/mtex#1677] [Ask Anything] [answered]
Hi everyone,

Let say I have 2 crystals with orientation in Bunge convention as below. How to calculate a misorientation (by Euler) about a certain axis (for example, <001>)? 

```
  Bunge Euler angles in degree
  phi1     Phi    phi2    Inv.
   360 20      30       0
   270 45      10       0
```

**Comment by ralfHielscher** [comment, marked as answer]
Can you explain what you mean by misorientation about a certain axis?

**Reply by trudator** [reply]
My question may be misunderstood. I have adjusted to a proper one. Let say that I have 2 orientations listed below. The misorientation basically just a phi_1 deviation of +/- 90 degree rotation about ND. However, the misorientation returns with an unexpected result. 

```
ori_1 = orientation.byEuler(180*degree,35*degree,45*degree,cs);
ori_2 = orientation.byEuler(90*degree,35*degree,45*degree,cs);
misorientation= inv(ori_2) * ori_1
```

Misorientation result:
```
  Bunge Euler angles in degree
     phi1     Phi    phi2    Inv.
  84.3227 47.8549 354.323       0
```

**Reply by ralfHielscher** [reply]
There are several aspects to consider:

 1. `inv(ori_2) * ori_1` computes the misorientation in crystal coordinates, i.e., from one crystal reference of crystal A into crystal reference frame from crystal B. There is no ND in these reference systems
 2. `ori_1 * inv(ori_2)` would compute the misorientation in specimen coordinates - however ignoring crystal symmetry. I'm not sure whether this is what you want to have.
 3. `axis(ori_1,ori_2)` gives the rotational axis in specimen, taking crystal symmetry into account. `angle(ori_1,ori_2)/degree` is the corresponding angle

Ralf.

**Reply by trudator** [reply]
The second bullet is exactly what I wanted. Thanks Ralf so much!

Just a quick follow-up question. Is there any difference between:

```
ori_1= orientation.byMiller([1 1 2],[1 1 0],cs)
ori_2 = orientation.byMiller([1 1 2],[1 1 1],cs)
axis(ori_1,ori_2)
```

and

```
ori_1= orientation.byMiller([1 1 2],[1 1 0],cs)
ori_2 = orientation.byMiller([1 1 2],[1 1 1],cs)
misorientation =  inv(ori_1) * ori_2
misorientation.axis
```

It generated two different results.

**Reply by ralfHielscher** [reply]
The first one is the misorientation axis in specimen symmetry, while the second one is the misorientation axis in crystal coordinates. 
Have a look at https://mtex-toolbox.github.io/orientation.axis.html

Ralf

**Reply by trudator** [reply]
Thanks Ralf!
